src/main/java/org/opentripplanner/routing/edgetype/TimetableResolver.java

```diff
@@ -50,6 +52,47 @@ public int compare(Timetable t1, Timetable t2) {
             return t1.serviceDate.compareTo(t2.serviceDate);
         }
     }
+    
+    /**
+     * Class to use as key in HashMap containing trip id and service date
+     */
+    protected class TripIdAndServiceDate {
```

Comment:
Yes, I thought about this as well, but since this class will probably be extended in the future (adding a feedId for multiple feed support and/or adding a startTime for frequency based trips support) I chose this way. Because I used the Objects.hash() and Objects.equals() methods, it will be very easy to extend the class.